src/s390x/zipl.rs

```diff
@@ -28,109 +27,54 @@ use walkdir::WalkDir;
 /// # Arguments
 /// * `boot` - path to boot partition mountpoint, i.e. smth like /boot
 /// * `disk` - smth like /dev/dasda
+/// * `firstboot_kargs` - enables ignition firstboot and adds provided kargs
 pub fn install_bootloader<P: AsRef<Path>>(
     boot: P,
-    disk: &str,
-    extra_kargs: Option<&str>,
+    disk: Option<&str>,
+    firstboot_kargs: Option<&String>,
 ) -> Result<()> {
     eprintln!("Installing bootloader");
-
-    let bls_config_path = get_bls_config_path(&boot)?;
-    let kernel_path = get_kernel_path(&boot)?;
-    let initramfs_path = get_initramfs_path(&boot)?;
-
-    let kargs = format!(
-        "{} ignition.firstboot {}",
-        get_boot_kargs(bls_config_path)?,
-        extra_kargs.unwrap_or("")
+    let boot = boot.as_ref();
+    let conffile = "/tmp/zipl.conf";
+    let data = format!(
+        "[defaultboot]\ndefaultauto\nprompt=1\ntimeout=5\nsecure=auto\ntarget={}\n",
+        boot.to_string_lossy()
     );
+    std::fs::write(conffile, &data).with_context(|| format!("writing {}: {}", conffile, data))?;
 
-    let status = Command::new("zipl")
-        .arg("-P")
-        .arg(kargs)
-        .arg("-i")
-        .arg(kernel_path.as_os_str())
-        .arg("-r")
-        .arg(initramfs_path.as_os_str())
-        .arg("--target")
-        .arg(boot.as_ref().as_os_str())
-        .arg("-n")
-        .stdout(Stdio::null())
-        .status()
-        .with_context(|| format!("failed to execute zipl on {}", disk))?;
-    if !status.success() {
-        bail!("couldn't install bootloader on {}", disk);
-    }
-
-    eprintln!("Updating re-IPL device");
-    let status = Command::new("chreipl")
-        .arg(disk)
-        .stdout(Stdio::null())
-        .status()
-        .with_context(|| format!("failed to execute chreipl on {}", disk))?;
-    if !status.success() {
-        bail!("couldn't set {} as boot device", disk);
+    // we have to explicitly add some options to bls and remove them after invocation of zipl
+    let extra = if let Some(kargs) = firstboot_kargs {
+        let mut extra = vec!["ignition.firstboot".to_string()];
+        extra.extend_from_slice(
+            &kargs
+                .split_whitespace()
+                .map(|s| s.to_string())
+                .collect::<Vec<String>>(),
+        );
+        Some(extra)
+    } else {
+        None
```

Comment:
Even if there are no firstboot kargs provided, we still need to append `ignition.firstboot`, no?

Reply:
i don't think so, during installation on s390x we have `firstboot_kargs`, so we add `ignition.firstboot`.

Reply:
But we're not guaranteed to have any `firstboot_kargs`.